### PR TITLE
TEST-#7076: Explicitly check for exceptions in `test_map_metadata.py`

### LINKED ISSUE #7076: Explicitly check for exceptions in `test_map_metadata.py`

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #7076 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/dataframe/test_map_metadata.py [hunk 1]
```diff
@@ -1385,13 +1405,22 @@ def comparator(df1, df2):
     for idx, value in enumerate(
         (pandas_df.to_numpy(), np.array([[1]] * 3), np.array([[1, 2, 3], [4, 5, 6]]))
     ):
+        raising_exceptions = None
+        if idx == 0:
+            raising_exceptions = ValueError(
+                "Expected a 1D array, got an array with shape (3, 2)"
+            )
+        elif idx == 2:
+            # Modin's exception message looks better, it's probably fine just leave it as is
```
Comment (YarShev) [inline modin/pandas/test/dataframe/test_map_metadata.py:1414]: Can you provide an example of the messages from Modin and pandas? Even if our message looks better, why should we not match pandas?

File: modin/pandas/test/dataframe/test_map_metadata.py [hunk 2]
```diff
@@ -1633,10 +1665,14 @@ def test___neg__(request, data):
...
 @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
-def test___invert__(data):
-    modin_df = pd.DataFrame(data)
-    pandas_df = pandas.DataFrame(data)
-    eval_general(modin_df, pandas_df, lambda df: ~df)
+def test___invert__(data, request):
+    raising_exceptions = None
+    if "float_nan_data" in request.node.callspec.id:
+        # Modin's exception message looks better, it's probably fine just leave it as is
+        raising_exceptions = False
```
Comment (YarShev) [inline modin/pandas/test/dataframe/test_map_metadata.py:1672]: same

File: modin/pandas/test/dataframe/test_map_metadata.py [hunk 3]
```diff
@@ -1648,13 +1684,6 @@ def test___invert___bool():
     df_equals(modin_result, pandas_result)
 
 
-def test___hash__():
```
Comment (YarShev) [inline modin/pandas/test/dataframe/test_map_metadata.py:1651]: Why removed?

File: modin/pandas/test/dataframe/test_map_metadata.py [hunk 4]
```diff
@@ -1312,18 +1328,15 @@ def test_insert(data):
...
-        value=lambda df: df[df.columns[0]],
-    )
+    if StorageFormat.get() != "Hdk":
+        # FIXME: https://github.com/modin-project/modin/issues/7027
+        eval_insert(
+            pd.DataFrame(columns=list("ab")),
+            pandas.DataFrame(columns=list("ab")),
+            col=lambda df: df.columns[0],
+            value=lambda df: df[df.columns[0]],
+            allow_duplicates=True,
```
Comment (YarShev) [inline modin/pandas/test/dataframe/test_map_metadata.py:1338]: Why added?

File: modin/pandas/test/dataframe/test_map_metadata.py (hunk 3 above)
Comment (anmyachev) [reply modin/pandas/test/dataframe/test_map_metadata.py:1651]: dataframes are no longer hashable.

```python
>>> import pandas as pd       
>>> hash(pd.DataFrame([1,2,3,4])) 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'DataFrame'
```

File: modin/pandas/test/dataframe/test_map_metadata.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/dataframe/test_map_metadata.py]: I created new issues. Let’s decide it separately?

File: modin/pandas/test/dataframe/test_map_metadata.py (hunk 4 above)
Comment (anmyachev) [reply modin/pandas/test/dataframe/test_map_metadata.py]: removed

**Pull request merged**